src/MultiFormats.Tests/MultiHashTest.cs

```diff
@@ -0,0 +1,558 @@
+            }
+        }
+
+        [TestMethod]
+        public void CheckMultiHash_Stream()
+        {
+            foreach (var v in TestVectors)
+            {
+                if (v.Ignore) continue;
+                var bytes = Encoding.UTF8.GetBytes(v.Input);
+                using (var ms = new MemoryStream(bytes, false))
+                {
+                    var mh = MultiHash.ComputeHash(ms, v.Algorithm);
+                    Assert.AreEqual(v.Output, mh.ToArray().ToHexString(), v.Algorithm);
+                }
+            }
+        }
+
+        [TestMethod]
+        public void IdentityHash()
+        {
+            var hello = Encoding.UTF8.GetBytes("Hello, world.");
+            var mh = MultiHash.ComputeHash(hello);
+            Assert.IsFalse(mh.IsIdentityHash);
+            CollectionAssert.AreNotEqual(hello, mh.Digest);
+
+            mh = MultiHash.ComputeHash(hello, "identity");
+            Assert.IsTrue(mh.IsIdentityHash);
+            CollectionAssert.AreEqual(hello, mh.Digest);
+
+            var mh1 = new MultiHash(mh.ToBase58());
+            Assert.AreEqual(mh, mh1);
+
+            mh = MultiHash.ComputeHash(hello, "id");
+            Assert.IsTrue(mh.IsIdentityHash);
+            CollectionAssert.AreEqual(hello, mh.Digest);
+
+            mh1 = new MultiHash(mh.ToBase58());
+            Assert.AreEqual(mh, mh1);
+        }
+
+        [TestMethod]
+        public void Binary()
+        {
+            var mh = new MultiHash("QmPZ9gcCEpqKTo6aq61g2nXGUhM4iCL3ewB6LDXZCtioEB");
+            Assert.AreEqual("sha2-256", mh.Algorithm.Name);
+            Assert.AreEqual(32, mh.Digest.Length);
+
+            var binary = mh.ToArray();
+            var mh1 = new MultiHash(binary);
+            Assert.AreEqual(mh.Algorithm.Name, mh1.Algorithm.Name);
+            CollectionAssert.AreEqual(mh.Digest, mh1.Digest);
+        }
+
+        [TestMethod]
+        public void JsonSerialization()
+        {
+            var a = new MultiHash("QmPZ9gcCEpqKTo6aq61g2nXGUhM4iCL3ewB6LDXZCtioEB");
+            var json = JsonConvert.SerializeObject(a);
+            Assert.AreEqual($"\"{a}\"", json);
+            var b = JsonConvert.DeserializeObject<MultiHash>(json);
+            Assert.AreEqual(a, b);
+            
+            json = JsonConvert.SerializeObject(null);
+            b = JsonConvert.DeserializeObject<MultiHash>(json);
+            Assert.IsNull(b);
+        }
+
+        [TestMethod]
+        public void CodeToName()
+        {
+            Assert.AreEqual("sha2-512", MultiHash.GetHashAlgorithmName(0x13));
+            ExceptionAssert.Throws<KeyNotFoundException>(
+                () => MultiHash.GetHashAlgorithmName(0xbadbad));
+        }
+
+        [TestMethod]
+        public void GetAlgorithmByName()
+        {
+            Assert.IsNotNull(MultiHash.GetHashAlgorithm());
+            Assert.IsNotNull(MultiHash.GetHashAlgorithm("sha2-512"));
+            var _ = ExceptionAssert.Throws<KeyNotFoundException>(() =>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)